vpr/src/device/rr_graph_view.h

```diff
@@ -96,6 +96,72 @@ class RRGraphView {
         return node_storage_.fan_in(node);
     }
 
+    /* Get the xlow of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_xlow(RRNodeId node) const {
+        return node_storage_.node_xlow(node);
+    }
+
+    /* Get the xhigh of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_xhigh(RRNodeId node) const {
+        return node_storage_.node_xhigh(node);
+    }
+
+    /* Get the ylow of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_ylow(RRNodeId node) const {
+        return node_storage_.node_ylow(node);
+    }
+
+    /* Get the yhigh of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_yhigh(RRNodeId node) const {
+        return node_storage_.node_yhigh(node);
+    }
+
+    /* Get the length of a routing resource node. This function is inlined for runtime optimization. */
+    /* node_length() only applies to CHANX or CHANY and is always a positive number*/
+    inline int node_length(RRNodeId node) const {
+        VTR_ASSERT(node_type(node) == CHANX || node_type(node) == CHANY);
+        int length = 1 + node_xhigh(node) - node_xlow(node) + node_yhigh(node) - node_ylow(node);
+        VTR_ASSERT_SAFE(length > 0);
+        return length;
+    }
+
+    /* Check if routing resource node is initialized. This function is inlined for runtime optimization. */
+    inline bool node_is_initialized(RRNodeId node) const {
+        return !((node_type(node) == NUM_RR_TYPES)
+                 && (node_xlow(node) == -1) && (node_ylow(node) == -1)
+                 && (node_xhigh(node) == -1) && (node_yhigh(node) == -1));
+    }
+
+    /* Check if two routing resource nodes are adjacent (must be a CHANX and a CHANY). This function is inlined for runtime optimization. */
+    inline bool nodes_are_adjacent(RRNodeId chanx_node, RRNodeId chany_node) const {
```

Comment:
Expand the comment to say this routine is used for error checking; it checks if two nodes are physically adjacent (could be connected) based on their geometry.  It does not check the routing edges to see if there are in fact possible to connect in the current routing graph.